db/log_reader.h

```diff
@@ -93,6 +93,10 @@ class Reader {
 
   uint64_t GetLogNumber() const { return log_number_; }
 
+  size_t end_of_buffer_offset() const {
```

Comment:
Inside log::Reader, the name, 'end_of_buffer_offset_' makes sense. However, if we expose this info as an API, should we consider another name that is easier to understand? User of this API may not even care what a 'buffer' refers to.